src/Xrefcheck/Scanners/Markdown.hs

```diff
@@ -340,24 +404,31 @@ nodeExtractInfo fp input@(Node _ _ nSubs) = do
            (DList.singleton $ Reference {rName, rPos, rLink, rAnchor})
            DList.empty
 
--- | Check if there is `ignore all` at the beginning of the file,
--- ignoring preceding comments if there are any.
-checkIgnoreAllFile :: [Node] -> Bool
-checkIgnoreAllFile nodes =
-  let isSimpleComment :: Node -> Bool
-      isSimpleComment node = isComment node && not (isIgnoreFile node)
-
-      mIgnoreFile = safeHead $ dropWhile isSimpleComment nodes
-  in maybe False isIgnoreFile mIgnoreFile
+-- | Check for global annotations, ignoring simple comments if there are any.
+checkGlobalAnnotations :: [C.Node] -> (Bool, [C.Node])
+checkGlobalAnnotations nodes = do
+  let (headerNodes, contentsNodes) = span isHeaderNode nodes
+      ignoreFile = any isIgnoreFile headerNodes
+  (ignoreFile, contentsNodes)
   where
-    isComment :: Node -> Bool
-    isComment = isJust . getCommentContent
+    isSimpleComment :: C.Node -> Bool
+    isSimpleComment node = do
+      let isComment = isJust $ getCommentContent node
+          isNotXrefcheckAnnotation = isNothing $ getXrefcheckContent node
+      isComment && isNotXrefcheckAnnotation
 
-    isIgnoreFile :: Node -> Bool
-    isIgnoreFile = (ValidMode IMAll ==) . getIgnoreMode
+    isIgnoreFile :: C.Node -> Bool
+    isIgnoreFile = (Just (IgnoreAnnotation IMAll) ==) . getAnnotation
```

Comment:
Maybe we can have a function like

`isGlobalAnnotation :: GetAnnotation -> Bool`

and use it here. Also, if this function has no wildcard pattern matchings, the compiler will ask us if we want any new annotation that we add in the future to be global or not.

Reply:
Currently there can be only one type of annotation, which is `IgnoreFile`. [Here](https://github.com/serokell/xrefcheck/pull/240) similar function is introduced, called `isHeaderNode`.